dashboard/app/models/school.rb

```
@@ -403,7 +458,11 @@ def self.merge_from_csv(filename, options = CSV_IMPORT_OPTIONS, update_existing
               state_cs_offerings_deleted_count += deleted_state_cs_offerings.count
             end
 
-            db_entry.update!(csv_entry) unless is_dry_run
+            begin
+              db_entry.update!(csv_entry)
+            rescue
```

Rescue the specific exception (`ActiveRecord::RecordNotUnique`)?

---

Or, if this is supposed to be a catchall for testing, maybe only rescue if we're in a dry run? Not even sure if a conditional rescue is possible...

---

I think this `errored_schools` entry will catch similar schools to the `duplicate_schools` (ie, those which hit a `ActiveRecord::RecordNotUnique` error) -- maybe think about how to combine these?

---

I think it makes sense to combine those and make it specific to the exception. I'd like it to error out for an unexpected thing that wasn't caught before.

---

Sounds good to me!